lib/src/main/java/com/otaliastudios/transcoder/source/DefaultDataSource.java

```diff
@@ -130,13 +130,18 @@ public long requestKeyFrameTimestamps() {
                 if (!keyFrameTimestamps.isEmpty() && sampleTime <= keyFrameTimestamps.get(keyFrameTimestamps.size() - 1)) {
                     Collections.sort(keyFrameTimestamps);
                 }
-                keyFrameTimestamps.add(sampleTime);
+                if (!keyFrameTimestamps.contains(sampleTime)) {
```

Comment:
this is unoptimal.

Reply:
instead of searching the whole list, now comparing with only last item of list, since it's ordered.